server/autolinkplugin/plugin.go

```diff
@@ -181,9 +174,29 @@ func (p *Plugin) ProcessPost(c *plugin.Context, post *model.Post) (*model.Post,
 			newText := origText
 
 			for _, link := range conf.Links {
-				if p.inScope(link.Scope, channelName, teamName) {
-					newText = link.Replace(newText)
+				if !p.inScope(link.Scope, channelName, teamName) {
+					continue
+				}
+
+				out := link.Replace(newText)
+				if out == newText {
+					continue
+				}
+
+				if !link.ProcessBotPosts {
+					if author == nil && authorErr == nil {
+						author, authorErr = p.API.GetUser(post.UserId)
+						if authorErr != nil {
+							p.API.LogError("failed to check if message for rewriting was send by a bot", "error", authorErr)
```

Comment:
Should we copy the deleted comment for below:
```suggestion
							// NOTE: Not sure how we want to handle errors here, we can either:
							// * assume that occasional rewrites of Bot messges are ok
							// * assume that occasional not rewriting of all messages is ok
							// Let's assume for now that former is a lesser evil and carry on.
							p.API.LogError("failed to check if message for rewriting was send by a bot", "error", authorErr)
```